conans/client/cmd/search.py

```diff
@@ -43,19 +43,52 @@ def search_packages(self, reference=None, remote_name=None, query=None, outdated
                 packages_pattern = String query with binary
                                    packages properties: "arch=x86 AND os=Windows"
         """
-        if remote_name:
-            remote = self._registry.remote(remote_name)
-            packages_props = self._remote_manager.search_packages(remote, reference, query)
-            ordered_packages = OrderedDict(sorted(packages_props.items()))
-            manifest = self._remote_manager.get_conan_manifest(reference, remote)
-            recipe_hash = manifest.summary_hash
-        else:
+        remote_ref = namedtuple('remote_ref', 'ordered_packages recipe_hash')
+        references = OrderedDict()
+        if not remote:
             packages_props = search_packages(self._client_cache, reference, query)
             ordered_packages = OrderedDict(sorted(packages_props.items()))
             try:
                 recipe_hash = self._client_cache.load_manifest(reference).summary_hash
             except IOError:  # It could not exist in local
                 recipe_hash = None
+
+            if outdated and recipe_hash:
+                ordered_packages = filter_outdated(ordered_packages, recipe_hash)
+
+            references[None] = remote_ref(ordered_packages, recipe_hash)
+            return references
+
+        if remote == 'all':
+            remotes = self._registry.remotes
+            # We have to check if there is a remote called "all"
+            # Deprecate: 2.0 can remove this check
+            if 'all' not in (r.name for r in remotes):
+                for remote in remotes:
+                    try:
+                        packages_props = self._remote_manager.search_packages(remote, reference, query)
+                        if packages_props:
+                            ordered_packages = OrderedDict(sorted(packages_props.items()))
+                            manifest = self._remote_manager.get_conan_manifest(reference, remote)
+                            recipe_hash = manifest.summary_hash
+
+                            if outdated and recipe_hash:
+                                ordered_packages = filter_outdated(ordered_packages, recipe_hash)
+
+                            references[remote.name] = remote_ref(ordered_packages, recipe_hash)
+                    except NotFoundException:
+                        continue
+                return references
+
+        # single remote
```

Comment:
This might benefit from some refactor, the method is a bit long and some small repetitions.
Not to do it now necessarily, could be an improvement in the future, it is good enough as is